packages/gateway/src/errors.js

```diff
@@ -0,0 +1,25 @@
+export class InvalidIpfsPathError extends Error {
+  /**
+   * @param {string} cid
+   */
+  constructor(cid) {
+    super(`invalid ipfs path: invalid path "/ipfs/${cid}/"`)
+    this.name = 'InvalidIpfsPathError'
+    this.status = 400
+    this.code = InvalidIpfsPathError.CODE
+  }
+}
+InvalidIpfsPathError.CODE = 'ERROR_INVALID_IPFS_PATH'
+
+export class InvalidUrlError extends Error {
+  /**
+   * @param {string} url
+   */
+  constructor(url) {
+    super(`invalid url: ${url}`)
```

Comment:
Probably don't need to echo the URL back...

```suggestion
  /**
   * @param {string} message
   */
  constructor(message = 'invalid URL') {
    super(message)
```